types/prop-types/index.d.ts

```diff
@@ -67,7 +67,7 @@ export const element: Requireable<ReactElementLike>;
 export const symbol: Requireable<symbol>;
 export const elementType: Requireable<ReactComponentLike>;
 export function instanceOf<T>(expectedClass: new (...args: any[]) => T): Requireable<T>;
-export function oneOf<T>(types: ReadonlyArray<T>): Requireable<T>;
+export function oneOf<T extends boolean | number | object | string | symbol>(types: ReadonlyArray<T>): Requireable<T>;
 export function oneOfType<T extends Validator<any>>(types: T[]): Requireable<NonNullable<InferType<T>>>;
```

Comment:
looks like this messes with `oneOfType` and makes it so inline declaration types don't get narrowed down:

inline:
![oneOfType-inline](https://user-images.githubusercontent.com/758244/90309926-a3458580-df0e-11ea-8797-c06aa6c343aa.png)

not inline:
![oneOfType-not-inline](https://user-images.githubusercontent.com/758244/90309928-a6407600-df0e-11ea-92eb-fff086050100.png)

We could change `Validator<any>` to `Validator<Literal>` here - but that would only work for 1 argument - i'm afraid i'm not knowledgable enough on typescript to solve this

closing the PR instead